.github/workflows/twister_dfu_nrf52840dk.yml

```diff
@@ -0,0 +1,138 @@
+  #
+  # export GOLIOTH_SYSTEM_SERVER_HOST=coap.golioth.dev
+  # export GOLIOTH_SYSTEM_CLIENT_PSK_ID=device_psk_id@ci
+  # export GOLIOTH_SYSTEM_CLIENT_PSK=device_psk
+  # export GOLIOTH_SAMPLE_WIFI_SSID=golioth-runner-xxx
+  # export GOLIOTH_SAMPLE_WIFI_PSK=password_for_wifi
+  # export GOLIOTH_DEVICE_NAME=nrf52840dk
+  # export GOLIOTHCTL_CONFIG=$HOME/.goliothctl-local.yaml
+  #
+  # For item 6, the file needs to have contents like (use base64 encoding):
+  #
+  # accesstoken: veryverylongtokenhash
+  # apikey: ""
+  # apiurl: https://api.golioth.dev
+  # config: ""
+  # projectid: ci
+  #
+  # NOTE: To store files as GitHub secrets, use base64 encoding:
+  #
+  # $ base64 -i --wrap=0 your_secret_file.yaml
+
+  twister_DFU:
+    runs-on: [self-hosted, has_twister_nrf52840dk]
+
+    container:
+      image: zephyrprojectrtos/ci:v0.26.1
+      volumes:
+        - /dev:/dev
+        - /opt/nrf-command-line-tools:/opt/nrf-command-line-tools
+        - /opt/SEGGER/JLink:/opt/SEGGER/JLink
+      options: --privileged
+
+    env:
+      ZEPHYR_SDK_INSTALL_DIR: /opt/toolchains/zephyr-sdk-0.16.0
+
+    defaults:
+      run:
+        shell: bash
+        working-directory: twister_run
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          path: twister_run/modules/lib/golioth
+
+      - name: Init and update west
+        run: |
+          mkdir -p .west
+          cat <<EOF > .west/config
+          [manifest]
+          path = modules/lib/golioth
+          file = .ci-west-zephyr.yml
+          EOF
+
+          west update -o=--depth=1 -n
+          west patch --apply
+
+      - name: Installing Python reqs
+        run: |
+          cd modules/lib/golioth/
+          pip install -r scripts/requirements-tests.txt
+          cd scripts/python/golioth
+          pip install .
+
+      - name: Format Runner Name
+        run: |
+          RUNNERNAME=$(echo ${{ runner.name }} | sed -e 's/-/_/g' -e 's/[a-z]/\U&/g')
+          echo "hil_runner=$RUNNERNAME" >> "$GITHUB_ENV"
+          echo "hil_runner_nrf52840dk=${RUNNERNAME}_NRF52840DK" >> $GITHUB_ENV
+          echo "hil_runner_nrf52840dk_port=${RUNNERNAME}_NRF52840DK_PORT" >> $GITHUB_ENV
+
+      - name: Copy credentials into container
+        run: |
+          echo ${{ secrets[env.hil_runner_nrf52840dk] }} | base64 -d > env_dfu_nrf52840dk.sh
+          echo ${{ secrets[env.hil_runner_nrf52840dk_port] }} | base64 -d > hardware-map.yaml
+          echo ${{ secrets.GOLIOTHCTL_YAML }} | base64 -d > .goliothctl-local.yaml
+
+      - name: Run Twister
+        run: |
+          source env_dfu_nrf52840dk.sh
+          cp .goliothctl-local.yaml ${HOME}/.
+          sed -i 's/CONFIG_SETTINGS=y/CONFIG_SETTINGS=n/' modules/lib/golioth/samples/dfu/prj.conf
+          PATH=$PATH:/opt/nrf-command-line-tools/bin
+          nrfjprog -i
+          zephyr/scripts/twister -vvv -p nrf52840dk_nrf52840 -T modules/lib/golioth/samples/dfu --device-testing --hardware-map hardware-map.yaml --west-flash=''
```

Comment:
We can provide `modules/lib/golioth/samples` to test all samples, not just `samples/dfu`. In that case it would be good to provide `-t goliothd`, as `goliothd` tag is used in all samples that require communication with Golioth backend.

Reply:
In recent discussion with @sam-golioth, we'd like to have cycle testing just for DFU. I'd like to merge this to focus only on the DFU sample for now and see how it performs is scheduled (an manual) runs over the next sprint, then open up a workflow that runs all samples.

Reply:
Once we move on running all samples, we'll want each test (i.e. each sample) to show up separately in the results in Github CI, so depending on how Twister/Github spit out the results, we may want to run them all separately anyway.